build_scripts/common/basic-ceph-server.sh

```diff
@@ -0,0 +1,209 @@
+
+# cleanup old build dir
+[ -d build ] && rm -rf build
+
+mkdir build
+cd build
+
+( cmake ../src -DCMAKE_BUILD_TYPE=Maintainer -DUSE_FSAL_GLUSTER=OFF -DUSE_FSAL_CEPH=ON -DUSE_FSAL_RGW=OFF -DUSE_DBUS=ON -DUSE_ADMIN_TOOLS=ON && make) || touch FAILED
+make install
+
+# dont vote if the subject of the last change includes the word "WIP"
+if ( git log --oneline -1 | grep -q -i -w 'WIP' )
+then
+    echo "Change marked as WIP, not posting result to GerritHub."
+    touch WIP
+fi
+
+# If failure found during build, return the status and skip proceeding
+# to ceph configuration
+
+
+# we accept different return values
+# 0 - SUCCESS + VOTE
+# 1 - FAILED + VOTE
+# 10 - SUCCESS + REPORT ONLY (NO VOTE)
+# 11 - FAILED + REPORT ONLY (NO VOTE)
+RET=0
+if [ -e FAILED ]
+then
+	exit ${RET}
+fi
+if [ -e WIP ]
+then
+	RET=$[RET + 10]
+	exit ${RET}
+fi
+
+# Create a virtual disk file (for OSD storage):
+truncate -s 35G /tmp/ceph-disk.img
+losetup -f /tmp/ceph-disk.img  # Attaches as a loop device (e.g., /dev/loop0)
+
+pvcreate /dev/loop0
+vgcreate ceph-vg /dev/loop0
+lvcreate -L 10G -n osd1 ceph-vg
+lvcreate -L 10G -n osd2 ceph-vg
+lvcreate -L 10G -n osd3 ceph-vg
+
+# Install and configure ceph cluster
+dnf install -y cephadm
+cephadm add-repo --release squid
+dnf install -y ceph
+cephadm bootstrap --mon-ip $(hostname -I | awk '{print $1}') --single-host-defaults --allow-fqdn-hostname
+ceph auth get client.bootstrap-osd -o /var/lib/ceph/bootstrap-osd/ceph.keyring
+
+# Attach the virtual disks
+ceph-volume lvm create --data /dev/ceph-vg/osd1
+ceph-volume lvm create --data /dev/ceph-vg/osd2
+ceph-volume lvm create --data /dev/ceph-vg/osd3
+
+# Verifying the disks
+lvdisplay
+lsblk
+ceph orch device ls
+
+# Now auto assign these lvms to the osd's
+ceph orch apply osd --all-available-devices
+
+# Wait for the osd's to be added
+
+echo "Waiting for at least one OSD to be ready..."
+TIMEOUT=300
+START_TIME=$(date +%s)
+while true; do
+    # Check if any OSD service exists and has at least one running OSD
+    OSD_STATUS=$(ceph orch ls --service-type osd --format json 2>/dev/null | \
+                 jq -r '.[0].status | select(.running != null) | .running >= 1')
+
+    # Check if the command succeeded and we got "true"
+    if [ "$OSD_STATUS" = "true" ]; then
+        echo "OSD is ready!"
+        break
+    fi
+
+    # Check timeout if set
+    if [ "$TIMEOUT" -gt 0 ]; then
+        CURRENT_TIME=$(date +%s)
+        ELAPSED=$((CURRENT_TIME - START_TIME))
+        if [ "$ELAPSED" -ge "$TIMEOUT" ]; then
+            echo "Timeout reached while waiting for OSD to be ready"
+            exit 1
+        fi
+    fi
+
+    sleep 5
+done
+
+# view the osd's
+ceph osd tree
+# Create a cephfs volume
+ceph fs volume create cephfs
+
+# Create subvolumegroup
+ceph fs subvolumegroup create cephfs ganeshagroup
+
+# Create subvolume
+ceph fs subvolume create cephfs nfs_subvol --group_name ganeshagroup --namespace-isolated
+
```

Comment:
Would it make sense to have a single script that would install and configure ceph?

This way we can reuse the same steps across other workflows.